src/PowerShellEditorServices/Services/Symbols/SymbolsService.cs

```diff
@@ -106,288 +107,178 @@ internal class SymbolsService
-
-            // Using a nested method here to get a bit more readability and to avoid roslynator
-            // asserting we should use a giant nested ternary here.
-            static string[] GetIdentifiers(string symbolName, SymbolType symbolType, CommandHelpers.AliasMap aliases)
-            {
-                if (symbolType is not SymbolType.Function)
-                {
-                    return new[] { symbolName };
-                }
-
-                if (!aliases.CmdletToAliases.TryGetValue(symbolName, out List<string> foundAliasList))
-                {
-                    return new[] { symbolName };
-                }
-
-                return foundAliasList.Prepend(symbolName)
-                    .Distinct(StringComparer.OrdinalIgnoreCase)
-                    .ToArray();
-            }
-
-            string[] allIdentifiers = GetIdentifiers(targetName, foundSymbol.SymbolType, aliases);
+            List<SymbolReference> symbols = new();
+            string[] allIdentifiers = GetIdentifiers(targetName, symbol.SymbolType, aliases);
 
             foreach (ScriptFile file in _workspaceService.GetOpenedFiles())
             {
                 foreach (string targetIdentifier in allIdentifiers)
                 {
-                    if (!file.References.TryGetReferences(targetIdentifier, out ConcurrentBag<IScriptExtent> references))
-                    {
-                        continue;
-                    }
+                    await Task.Yield();
+                    cancellationToken.ThrowIfCancellationRequested();
 
-                    foreach (IScriptExtent extent in references.OrderBy(e => e.StartOffset))
+                    _ = file.References.TryGetReferences(targetIdentifier, out ConcurrentBag<SymbolReference>? references);
+                    if (references is null)
                     {
-                        SymbolReference reference = new(
-                            SymbolType.Function,
-                            foundSymbol.SymbolName,
-                            extent);
-
-                        try
-                        {
-                            reference.SourceLine = file.GetLine(reference.ScriptRegion.StartLineNumber);
-                        }
-                        catch (ArgumentOutOfRangeException e)
-                        {
-                            reference.SourceLine = string.Empty;
-                            _logger.LogException("Found reference is out of range in script file", e);
-                        }
-                        reference.FilePath = file.FilePath;
-                        symbolReferences.Add(reference);
+                        continue;
                     }
 
-                    await Task.Yield();
-                    cancellationToken.ThrowIfCancellationRequested();
+                    symbols.AddRange(references);
                 }
             }
 
-            return symbolReferences;
+            return symbols;
```

Comment:
```suggestion
            return symbols.Where((reference) =>
                    symbol.SymbolType == reference.SymbolType ||
                    (symbol.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type
                        && reference.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type) ||
                    (symbol.SymbolType is SymbolType.EnumMember or SymbolType.Property
                        && reference.SymbolType is SymbolType.EnumMember or SymbolType.Property)
                );
```

Fix code lense reference count